internal/runtime/runtime.go

```diff
@@ -0,0 +1,127 @@
+package runtime
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/saucelabs/saucectl/internal/msg"
+	"golang.org/x/mod/semver"
+)
+
+const NodeRuntime = "nodejs"
+
+var namingMap = map[string]string{
+	NodeRuntime: "Node.js",
+}
+
+// Runtime represents runtime details on the VM.
+type Runtime struct {
+	RuntimeName    string
+	RuntimeAlias   []string
+	RuntimeVersion string
+	EOLDate        time.Time
+	Default        bool
+	Extra          map[string]string
+}
+
+// SelectNode selects the appropriate Node.js runtime from a list of runtimes.
+// It supports full SemVer matching, alias resolution, and fuzzy matching for major or major.minor versions.
+// `version` is expected to always start with "v".
+func SelectNode(runtimes []Runtime, version string) (Runtime, error) {
+	rts := filterNodeRuntimes(runtimes)
+	if !semver.IsValid(version) {
+		// If version is not a valid SemVer, check if it's using an alias (e.g., "lts" or code name).
+		res := findRuntimeByAlias(rts, version)
+		if res.RuntimeName != "" {
+			return res, nil
+		}
```

Comment:
(style) I think it'd be more canonical go if `findRuntimeByAlias` were to return the runtime and error. The error path is easier to identify that way.